javalib/src/main/scala/java/io/InputStreamReader.scala

```diff
@@ -65,6 +65,7 @@ class InputStreamReader(private[this] var in: InputStream,
 
   def read(cbuf: Array[Char], off: Int, len: Int): Int = {
     ensureOpen()
+    if (endOfInput) return -1
```

Comment:
My experiments show that this test should actually come after the `IndexOutOfBoundsException` test *and* the `len == 0` test:
```scala
$ scala
Welcome to Scala version 2.10.0 (Java HotSpot(TM) 64-Bit Server VM, Java 1.6.0_39).
Type in expressions to have them evaluated.
Type :help for more information.

scala> import java.io._
import java.io._

scala> val data = "Lorem ipsum".getBytes()
data: Array[Byte] = Array(76, 111, 114, 101, 109, 32, 105, 112, 115, 117, 109)

scala> new InputStreamReader(new ByteArrayInputStream(data))
res0: java.io.InputStreamReader = java.io.InputStreamReader@686baa51

scala> val bytes = new Array[Char](11)
bytes: Array[Char] = Array(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) 
scala> streamReader.read(bytes)
<console>:12: error: not found: value streamReader
              streamReader.read(bytes)
              ^

scala> val streamReader = res0
streamReader: java.io.InputStreamReader = java.io.InputStreamReader@686baa51

scala> streamReader.read(bytes)
res2: Int = 11

scala> streamReader.read(bytes)
res3: Int = -1

scala> streamReader.read(new Array(0))
<console>:14: error: overloaded method value read with alternatives:
  (Array[Char])Int <and>
  (java.nio.CharBuffer)Int
 cannot be applied to (Array[Nothing])
              streamReader.read(new Array(0))
                           ^

scala> streamReader.read(new Array[Byte](0))
<console>:14: error: overloaded method value read with alternatives:
  (Array[Char])Int <and>
  (java.nio.CharBuffer)Int
 cannot be applied to (Array[Byte])
              streamReader.read(new Array[Byte](0))
                           ^

scala> streamReader.read(new Array[Char](0))
res6: Int = 0

scala> streamReader.read(new Array[Char](5))
res7: Int = -1

scala> streamReader.read(new Array[Char](5), 11, 3)
java.lang.IndexOutOfBoundsException
        at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:131)
        at java.io.InputStreamReader.read(InputStreamReader.java:167)
        at .<init>(<console>:14)
        at .<clinit>(<console>)
        at .<init>(<console>:7)
        at .<clinit>(<console>)
        at $print(<console>)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:731)
        at scala.tools.nsc.interpreter.IMain$Request.loadAndRun(IMain.scala:980)
        at scala.tools.nsc.interpreter.IMain.loadAndRunReq$1(IMain.scala:570)
        at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:601)
        at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:565)
        at scala.tools.nsc.interpreter.ILoop.reallyInterpret$1(ILoop.scala:745)
        at scala.tools.nsc.interpreter.ILoop.interpretStartingWith(ILoop.scala:790)
        at scala.tools.nsc.interpreter.ILoop.command(ILoop.scala:702)
        at scala.tools.nsc.interpreter.ILoop.processLine$1(ILoop.scala:566)
        at scala.tools.nsc.interpreter.ILoop.innerLoop$1(ILoop.scala:573)
        at scala.tools.nsc.interpreter.ILoop.loop(ILoop.scala:576)
        at scala.tools.nsc.interpreter.ILoop$$anonfun$process$1.apply$mcZ$sp(ILoop.scala:867)
        at scala.tools.nsc.interpreter.ILoop$$anonfun$process$1.apply(ILoop.scala:822)
        at scala.tools.nsc.interpreter.ILoop$$anonfun$process$1.apply(ILoop.scala:822)
        at scala.tools.nsc.util.ScalaClassLoader$.savingContextLoader(ScalaClassLoader.scala:135)
        at scala.tools.nsc.interpreter.ILoop.process(ILoop.scala:822)
        at scala.tools.nsc.MainGenericRunner.runTarget$1(MainGenericRunner.scala:83)
        at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:96)
        at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:105)
        at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
```